flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/ComponentMetricGroup.java

```diff
@@ -33,19 +36,42 @@
  * group could for example include the task attempt number (more fine grained identification), or
  * exclude it (for continuity of the namespace across failure and recovery).
  */
-public abstract class ComponentMetricGroup extends AbstractMetricGroup {
+public abstract class ComponentMetricGroup<P extends AbstractMetricGroup> extends AbstractMetricGroup<P> {
```

Comment:
I'm not sure what you mean, but `JobMetricGroup` inherits from `ComponentMetricGroup`.